src/EnergyPlus/Autosizing/CoolingCapacitySizing.cc

```diff
@@ -275,10 +275,30 @@ Real64 CoolingCapacitySizer::size(EnergyPlusData &state, Real64 _originalValue,
                     auto &thisAirloopDOAS = this->airloopDOAS[this->outsideAirSys(this->curOASysNum).AirLoopDOASNum];
                     DesVolFlow = thisAirloopDOAS.SizingMassFlow / state.dataEnvrn->StdRhoAir;
                     CoilInTemp = thisAirloopDOAS.SizingCoolOATemp;
-                    if (thisAirloopDOAS.m_FanIndex > -1 && thisAirloopDOAS.FanBlowTroughFlag &&
-                        thisAirloopDOAS.m_FanTypeNum == SimAirServingZones::CompType::Fan_System_Object) {
-                        int FanIndex = thisAirloopDOAS.m_FanIndex;
-                        Real64 DeltaT = state.dataHVACFan->fanObjs[FanIndex]->getFanDesignTemperatureRise(state);
+                    Real64 DeltaT = 0.0;
+                    if (thisAirloopDOAS.m_FanIndex > -1 && thisAirloopDOAS.FanBeforeCoolingCoilFlag) {
+                        if (thisAirloopDOAS.m_FanTypeNum == SimAirServingZones::CompType::Fan_ComponentModel) {
+                            Fans::FanInputsForDesHeatGain(state,
+                                                          thisAirloopDOAS.m_FanIndex,
+                                                          this->deltaP,
+                                                          this->motEff,
+                                                          this->totEff,
+                                                          this->motInAirFrac,
+                                                          this->fanShaftPow,
+                                                          this->motInPower,
+                                                          this->fanCompModel);
+                            FanCoolLoad = this->fanShaftPow + (this->motInPower - this->fanShaftPow) * this->motInAirFrac;
+                            this->dataFanEnumType = DataAirSystems::StructArrayLegacyFanModels;
+                        } else if (thisAirloopDOAS.m_FanTypeNum == SimAirServingZones::CompType::Fan_System_Object) {
+                            state.dataHVACFan->fanObjs[thisAirloopDOAS.m_FanIndex]->FanInputsForDesignHeatGain(
+                                state, this->deltaP, this->motEff, this->totEff, this->motInAirFrac);
+                            Real64 fanPowerTot = (DesVolFlow * this->deltaP) / this->totEff;
+                            FanCoolLoad = this->motEff * fanPowerTot + (fanPowerTot - this->motEff * fanPowerTot) * this->motInAirFrac;
+                            this->dataFanEnumType = DataAirSystems::ObjectVectorOOFanSystemModel;
+                        }
+                        this->dataFanIndex = thisAirloopDOAS.m_FanIndex;
+                        Real64 CpAir = Psychrometrics::PsyCpAirFnW(state.dataLoopNodes->Node(thisAirloopDOAS.m_FanInletNodeNum).HumRat);
+                        DeltaT = FanCoolLoad / (thisAirloopDOAS.SizingMassFlow * CpAir);
                         CoilInTemp += DeltaT;
                     }
```

Comment:
Where is the draw thru fan accounted for? For draw thru the coil outlet temp would need to be lowered by fan delta T.

Reply:
@rraustad I added fan heat to the cooling coil, if a fan is placed before the cooling coil. Do we need to add heat heat to the cooling coil, if a fan is placed after the cooling coil?

Reply:
The OA coil will be sized to meet the Precool temp which is at the outlet of the fan and coils. If a blow thru fan, fan heat is added to OA temp at the inlet of the coil. If draw thru, fan heat is subtracted from Precool temp and used as the coil outlet temp used for sizing so that after the fan the temperature will equal Precool temp.

    Sizing:System,
      11.0,                    !- Precool Design Temperature {C}